client/src/components/shared/OnboardingStepper/OnboardingStepper.tsx

```diff
@@ -0,0 +1,102 @@
+import cx from 'classnames';
+import { motion } from 'framer-motion';
+import React, { ReactNode, useMemo, useState } from 'react';
+import { useTranslation } from 'react-i18next';
+
+import Img from 'components/ui/Img';
+import Svg from 'components/ui/Svg';
+import Tooltip from 'components/ui/Tooltip';
+import useMediaQuery from 'hooks/helpers/useMediaQuery';
+import useOnboardingSteps from 'hooks/helpers/useOnboardingSteps';
+import useUserTOS from 'hooks/queries/useUserTOS';
+import useOnboardingStore from 'store/onboarding/store';
+import { four, one, three, two } from 'svg/onboardingStepper';
+
+import styles from './OnboardingStepper.module.scss';
+
+const OnboardingStepper = (): ReactNode => {
+  const { t } = useTranslation('translation', { keyPrefix: 'components.shared.onboardingStepper' });
+  const { setIsOnboardingModalOpen, lastSeenStep } = useOnboardingStore(state => ({
+    isOnboardingModalOpen: state.data.isOnboardingModalOpen,
+    lastSeenStep: state.data.lastSeenStep,
+    setIsOnboardingModalOpen: state.setIsOnboardingModalOpen,
+  }));
+
+  const { isDesktop } = useMediaQuery();
+  const { data: isUserTOSAccepted } = useUserTOS();
+  const [isUserTOSAcceptedInitial] = useState(isUserTOSAccepted);
+  const stepsToUse = useOnboardingSteps(isUserTOSAcceptedInitial);
+  const cxcy = 28;
+  const viewBox = '0 0 56 56';
+  const numberOfSteps = stepsToUse.length;
+
+  const svgNumber = useMemo(() => {
+    if (lastSeenStep === 1) {
+      return one;
+    }
+    if (lastSeenStep === 2) {
+      return two;
+    }
+    if (lastSeenStep === 3) {
+      return three;
+    }
+    return four;
+  }, [lastSeenStep]);
+
+  return (
+    <motion.div
+      key="OnboardingStepper"
+      animate={{ bottom: isDesktop ? 48 : 116, opacity: 1, right: isDesktop ? 48 : 24 }}
+      className={styles.root}
+      data-test="OnboardingStepper"
+      exit={{ bottom: isDesktop ? 24 : 92, opacity: 0, right: isDesktop ? 48 : 24 }}
+      initial={{ bottom: isDesktop ? 24 : 92, opacity: 0, right: isDesktop ? 48 : 24 }}
```

Comment:
Instead of that I would declare an object and spread it. IMHO it's much easier to understand differences at a glance.
```
const animationProps = isDesktop ? { animate, exit, initial } : { animate, exit, initial }
...
{...animationProps}
```